### DISCUSSION TITLE
Apache-skywalk-apm-8.9.1, 'storage-elasticsearch-plugin' Can I enable compression if I write data to ElasticSearch?

### DISCUSSION THREAD

**Discussion by peoplewhoworkhard** [apache/skywalking#10794] [Q&A] [answered]
Apache-skywalk-apm-8.9.1, 'storage-elasticsearch-plugin' Can I enable compression if I write data to ElasticSearch?
---
Our skywalkingOAP cluster service and Elastic cluster service are not in the same machine room, and data transmission across idc will cause a large bandwidth consumption. I wonder if compression transmission can be performed on skywalkingOAP aggregated data when it is stored in elasticsearch. Thanks for the answer. It's very urgent. 
Thank you very much!
Thank you very much!
Thank you very much!

**Comment by wu-sheng** [comment, marked as answer]
The simple answer is, don't put OAP and DBs in different data centers. No matter for the network bandwidth cost or huge latency, that is not good for product use.

**Reply by peoplewhoworkhard** [reply]
First of all, thank the author for your busy schedule to answer our questions! 

Yes, we know that such a solution causes bandwidth usage and kafka consumption latency issues. The way we've gone about it so far is that the skywalkingOAP service and ElasticSearch are not in the same room (oap in Beijing, elastic in Hong Kong), probably because there was a shortage of machine resources. The result is that the bandwidth of all four dedicated lines is full. However, for now we want to try and compress and store the data written to ElasticSearch without changing the existing solution and see if or how much we can reduce the bandwidth footprint.

**Reply by peoplewhoworkhard** [reply]
The skywalking version we use is Apap-Skywalk-apm-8.9.1. If I want to compress my oap aggregated data and send it to ElasticSearch, where do I go to change the code?
![图片](https://github.com/apache/skywalking/assets/44570840/139cb895-826c-4170-8c76-6b290cd656ae)
![图片](https://github.com/apache/skywalking/assets/44570840/6f813805-7f70-4c76-85d6-4109a56fdc05)

**Reply by wu-sheng** [reply]
It would not be much AFAIK. One day(most likely in 2024), BanyanDB could do remote sync for a clone database. That mechanism will consider a high compression rate as it is designed for cross data-centers.
FYI @hanahmily

**Reply by wu-sheng** [reply]
About ElasticSearch compression, I am afraid I don't know. You need to dig the Elastic docs by yourselves. In any circumstance, we don't recommend OAP and ES in different physical places.

**Reply by peoplewhoworkhard** [reply]
Ok, thanks a lot. 
Skywalking is really great, thanks for your contribution.
